File: oracles/C Oracles/07 - Fibonacci/FibonacciTest.c
#include "FibonacciTest.h"

void test1(void) {

	unsigned char n = 0;

	CU_ASSERT_EQUAL(0, fibonacci(n));

}

void test2(void) {

	unsigned char n = 1;

	CU_ASSERT_EQUAL(1, fibonacci(n));

}

void test3(void) {

	unsigned char n = 2;

	CU_ASSERT_EQUAL(1, fibonacci(n));

}

void test4(void) {

	unsigned char n = 4;

	CU_ASSERT_EQUAL(3, fibonacci(n));

}

int main(void) {

	CU_pSuite pSuite = NULL;

	if (CUE_SUCCESS != CU_initialize_registry())
		return CU_get_error();

	pSuite = CU_add_suite("CUnit Suite", NULL, NULL);
	if (NULL == pSuite) {
		CU_cleanup_registry();
		return CU_get_error();
	}

	if ((NULL == CU_add_test(pSuite, "test1", test1)) ||
		(NULL == CU_add_test(pSuite, "test2", test2)) ||
		(NULL == CU_add_test(pSuite, "test3", test3)) ||
		(NULL == CU_add_test(pSuite, "test4", test4))
	) {
		CU_cleanup_registry();
		return CU_get_error();
	}

	CU_automated_run_tests();

	CU_cleanup_registry();

	return CU_get_error();

}

File: progtest-web/resources/oracles/Java/Fibonacci/07 - Fibonacci/FibonacciTest.java
import static org.junit.Assert.*;

import org.junit.Test;

public class FibonacciTest {

	@Test
	public void test1() {

		int n = 0;

		assertEquals(0, Fibonacci.fibonacci(n));

	}

	@Test
	public void test2() {

		int n = 1;

		assertEquals(1, Fibonacci.fibonacci(n));

	}

	@Test
	public void test3() {

		int n = 2;

		assertEquals(1, Fibonacci.fibonacci(n));

	}

	@Test
	public void test4() {

		int n = 4;

		assertEquals(3, Fibonacci.fibonacci(n));

	}

	@Test
	public void test5() {

		new Fibonacci();

	}

}
